aws-servicecatalog-serviceaction/src/main/java/software/amazon/servicecatalog/serviceaction/ActionController.java

```diff
@@ -79,6 +84,26 @@ public DescribeServiceActionResponse describeServiceAction(final String id) {
         return proxy.injectCredentialsAndInvokeV2(request, scClient::describeServiceAction);
     }
 
+    public List<String> listServiceAction() {
+        String pageToken = null;
+        List<String> serviceActionIds = new ArrayList<>();
+        do {
+            final ListServiceActionsRequest request = ListServiceActionsRequest
+                    .builder()
+                    .pageToken(pageToken)
+                    .build();
+            logger.log(LIST_SERVICE_ACTIONS_LOG);
+            ListServiceActionsResponse response = proxy.injectCredentialsAndInvokeV2(request, scClient::listServiceActions);
+            List<ServiceActionSummary> serviceActionSummaries = response.serviceActionSummaries();
+            pageToken = response.nextPageToken();
+            for (ServiceActionSummary serviceActionSummary: serviceActionSummaries) {
+                serviceActionIds.add(serviceActionSummary.id());
+            }
+        } while(!StringUtils.isNullOrEmpty(pageToken));
```

Comment:
Can we benefit from SDK2.0 to have an automated paginator or is this not available to service catalog APIs? See for details: https://aws.amazon.com/blogs/developer/auto-pagination-feature-in-java-sdk-2-0/

Reply:
Added automated paginator in new commit